source/h2_connection.c

```diff
@@ -772,6 +778,41 @@ int s_decoder_on_headers_end(
     return AWS_OP_SUCCESS;
 }
 
+int s_decoder_on_push_promise(uint32_t stream_id, uint32_t promised_stream_id, void *userdata) {
+    struct aws_h2_connection *connection = userdata;
+    AWS_ASSERT(connection->base.client_data); /* decoder enforces that server cannot receive PUSH_PROMISE */
```

Comment:
so we're going to crash a server process because of a misbehaving client? It seems better behavior here would be to process the error and quick kill the stream/connection.

Reply:
That particular bit of error-handling is done in h2_decoder.c, before we get out here to the connection. We do as much error checking as we possibly can in the decoder.

I was hoping the assert and comment would make that clear, but I guess it was confusing. I'll rephrase the comment to be more clear